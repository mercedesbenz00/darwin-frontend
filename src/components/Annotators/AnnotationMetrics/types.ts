import { MembershipPayload, DatasetAnnotationReportPayload } from '@/store/types'

export type AnnotationDataRange = 'total' | 'month' | 'week' | 'day'
export type AnnotationDataGranularity = 'day' | 'hour'

export type ChartDimension =
  'annotationTime' |
  'imagesAnnotated' |
  'avgTimePerAnnotation' |
  'annotationsCreated' |
  'reviewPassRate'

export type AdaptedDataPoint =
  {[k in Exclude<ChartDimension, 'reviewPassRate' | 'avgTimePerAnnotation'>]: number } &
  {[k in Extract<ChartDimension, 'reviewPassRate' | 'avgTimePerAnnotation'>]: number | null} &
  { userId: number | null, date: string }

export type AggregatedDataPoint = Pick<
    DatasetAnnotationReportPayload,
    | 'annotation_time'
    | 'annotations_approved'
    | 'annotations_created'
    | 'date'
    | 'images_annotated'
    | 'images_approved'
    | 'images_rejected'
    | 'user_id'
  >

export type AnnotationDimensionData = {
  annotationTime: number,
  annotationsCreated: number,
  avgTimePerAnnotation: number | null,
  imagesAnnotated: number,
  reviewPassRate: number | null
}

export type AnnotationChartData = {
  data: AdaptedDataPoint[]
  dimensions: AnnotationDimensionData
  member?: MembershipPayload
  visible: boolean
}

export type AnnotationBarChartData = {
  data: AggregatedDataPoint[]
  member?: MembershipPayload
}

/**
 * Format for a single item to be shown in a chart tooltip.
 *
 * A tooltip will contain multiple items
 */
export type ChartTooltipData = { member: MembershipPayload | null, value: string}
