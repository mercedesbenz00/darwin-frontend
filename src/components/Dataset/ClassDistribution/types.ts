import { AnnotationClassPayload, AnnotationTypePayload } from '@/store/types'

export type AnnotationTypeCount = {
  annotationType: AnnotationTypePayload
  total: number
}

export type ClassDistributionTab = 'instances' | 'files'

export enum DistributionStatus {
  VERY_OVERREPRESENTED = 'very-overrepresented',
  OVERREPRESENTED = 'overrepresented',
  BALANCED = 'balanced',
  UNDERREPRESENTED = 'underrepresented',
  VERY_UNDERREPRESENTED = 'very-underrepresented',

  VERY_LOW_DATA = 'very-low-data',
  LOW_DATA = 'low-data',
  MEDIUM_LOW_DATA = 'medium-low-data'
}

export type ClassOverallDistribution = {
  annotationClass: AnnotationClassPayload,
  countByImages: number
  countByInstances: number
  status: DistributionStatus
}

export type ClassDistributionDimension = keyof Pick<
  ClassOverallDistribution, 'countByImages' | 'countByInstances'
>

export type ClassIndividualDistribution = {
  annotationClass: AnnotationClassPayload,
  count: number
  status: DistributionStatus
}
