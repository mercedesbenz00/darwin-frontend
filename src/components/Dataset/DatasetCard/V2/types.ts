import { DatasetPayload } from '@/store/types'

export type DatasetCardProps = {
  dataset: DatasetPayload
  selectable: boolean
  disabled?: boolean
}
