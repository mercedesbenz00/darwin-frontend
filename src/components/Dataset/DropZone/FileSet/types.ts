import { UploadFile } from '@/store/modules/datasetUpload/types'

export const THUMB_LIMIT = 3

export type FilesByCategory = {
  images: UploadFile[],
  videos: UploadFile[],
  others: UploadFile[]
}
