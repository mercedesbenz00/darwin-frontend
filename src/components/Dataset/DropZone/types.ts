import { UploadFile } from '@/store/modules/datasetUpload/types'

export type UploadFileSet = {
  id: number
  files: UploadFile[]
}

export { UploadFile } from '@/store/modules/datasetUpload/types'
