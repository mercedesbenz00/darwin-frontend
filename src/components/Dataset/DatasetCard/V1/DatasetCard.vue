<template>
  <component
    :is="tag"
    class="dataset-card"
    :class="{ 'dataset-card--selected': selected }"
    :to="route"
    @click.stop="$emit('click')"
    @dblclick.stop="$emit('dblclick')"
  >
    <div class="dataset-card__upside">
      <div class="dataset-card__upside__row dataset-card__upside__row--top">
        <div class="dataset-card__img">
          <img
            v-if="thumbnails[0]"
            v-lazy="thumbnails[0]"
          >
        </div>
        <div class="dataset-card__img">
          <img
            v-if="thumbnails[1]"
            v-lazy="thumbnails[1]"
          >
        </div>
        <div class="dataset-card__img">
          <img
            v-if="thumbnails[2]"
            v-lazy="thumbnails[2]"
          >
        </div>
      </div>
      <div class="dataset-card__upside__row">
        <div class="dataset-card__img">
          <img
            v-if="thumbnails[3]"
            v-lazy="thumbnails[3]"
          >
        </div>
        <div class="dataset-card__img">
          <img
            v-if="thumbnails[4]"
            v-lazy="thumbnails[4]"
          >
        </div>
        <div class="dataset-card__img">
          <img
            v-if="thumbnails[5]"
            v-lazy="thumbnails[5]"
          >
        </div>
      </div>

      <transition
        v-if="!disableMenu"
        name="fade"
      >
        <div class="dataset-card__overlay">
          <div
            class="dataset-card__overlay__more"
            @click.stop.prevent="toggleMenu"
          >
            <img
              src="/static/imgs/more-white-icon.svg"
              class="dataset-card__overlay__more__img"
            >
          </div>
        </div>
      </transition>
      <div
        v-if="menu"
        v-click-outside="hideMenu"
        class="dataset-card__menu"
        :class="{'dataset-card__menu--right': menuAlign === 'right'}"
      >
        <div
          v-if="$can('archive_dataset', { subject: 'dataset', resource: data })"
          class="dataset-card__menu__item"
          @click.stop.prevent="onDelete"
        >
          Delete
        </div>
        <div
          v-else
          class="dataset-card__menu__item dataset-card__menu__item--disabled"
          @click.stop.prevent="notifyCannotDelete"
        >
          Delete
        </div>
      </div>
      <div
        v-if="data.public"
        class="dataset-card__status"
      >
        <img
          src="/static/imgs/dataset-status-open.svg"
          class="dataset-card__status-icon"
        >
        <div class="dataset-card__status-text">
          Open
        </div>
      </div>
    </div>
    <div class="dataset-card__downside">
      <div class="dataset-card__name">
        {{ data.name }}
      </div>
      <div class="dataset-card__stats">
        <div>
          <strong>{{ data.num_images + data.num_videos }}</strong> Files
        </div>
        <div>
          <strong>{{ data.num_classes }}</strong> Classes
        </div>
        <div>
          <strong>{{ progressPercentage }}%</strong> Complete
        </div>
      </div>
    </div>
    <div
      v-tooltip="{
        content: progressTooltip,
        placement: 'top',
        delay: { show: 300 },
        classes: 'tooltip--dataset-card'
      }"
      class="dataset-card__progress__bar"
    >
      <div
        class="dataset-card__progress__bar__filled"
        :style="{ width: progressPercentage + '%' }"
      />
    </div>
  </component>
</template>

<script lang = "ts">
import { Component, Prop, Vue } from 'vue-property-decorator'

import CheckBox from '@/components/Common/CheckBox/V1/CheckBox.vue'
import { DatasetPayload } from '@/store/types'
import { notifyErrorByCode } from '@/utils'

@Component({ name: 'dataset-card', components: { CheckBox } })
export default class DatasetCard extends Vue {
  @Prop({ required: true })
  data!: DatasetPayload

  @Prop({ default: 'normal' })
  menuAlign!: string

  @Prop({ default: false, type: Boolean })
  disableMenu!: boolean

  @Prop({ default: false, type: Boolean })
  selectable!: boolean

  @Prop({ default: false, type: Boolean })
  selected!: boolean

  menu: boolean = false

  get tag (): 'router-link' | 'div' {
    if (this.selectable) { return 'div' }
    return 'router-link'
  }

  get progressPercentage (): number {
    return Math.round((this.data.progress || 0) * 100)
  }

  get route (): string {
    if (this.data.version === 1) {
      return `/datasets/${this.data.id}/dataset-management`
    } else {
      return `/datasets/${this.data.id}/dataset-management?not_statuses=archived`
    }
  }

  get progressTooltip (): string {
    const { num_images: images, num_videos: videos, num_classes: classes } = this.data
    if (!images || !videos) { return '' }
    return `${images + videos} Files, ${classes} Classes, ${this.progressPercentage}% Complete`
  }

  get thumbnails (): string[] {
    return this.data.thumbnails ? this.data.thumbnails : []
  }

  toggleMenu (): void {
    this.menu = !this.menu
  }

  hideMenu (): void {
    this.menu = false
  }

  onDelete (): void {
    this.hideMenu()
    this.$emit('delete', this.data)
  }

  /**
   * "Delete" option in the dataset card dropdown is clickable
   * even when the user is not allowed to delete the dataset
   * and the appearance is that of a disabled control, but
   * instead of performing a backend request, it just notifies
   * why the user cannot delete the dataset.
   */
  notifyCannotDelete (): void {
    this.hideMenu()
    notifyErrorByCode(this.$store, 'DATASET_DELETE_NOT_AUTHORIZED')
  }
}
</script>

<!-- eslint-disable vue-scoped-css/enforce-style-type -->
<style lang="scss">
.tooltip--dataset-card {
  &[x-placement^="top"] {
    margin-bottom: 7px;
  }
}
</style>

<style lang="scss" scoped>
.dataset-card {
  @include col;
  display: inline-flex;
  height: 270px;
  border-radius: 7px;
  border: 3px solid transparent;
  position: relative;
  cursor: pointer;

  &:hover {
    .dataset-card__overlay {
      display: flex;
    }
  }

  &:not(:hover) {
    .dataset-card__overlay {
      display: none;
    }
  }

  .dataset-card--selected {
    .dataset-card__overlay {
      display: flex;
    }
  }
}

.dataset-card--selected {
  border-color: $colorFeatherLight;
}

.dataset-card__upside {
  width: 100%;
  height: 180px;
  border-radius: 5px 5px 0 0;
  position: relative;
}

.dataset-card__upside__row {
  width: 100%;
  height: 50%;
  @include row--center;
}

.dataset-card__upside__row--top {
  border-radius: 5px 5px 0 0;
  overflow: hidden;
}

.dataset-card__img {
  flex: 1 1 auto;
  height: 100%;
  background: #FAFCFE;
}

.dataset-card__overlay {
  position: absolute;
  @include fullsize;
  border-radius: 5px 5px 0 0;
  padding-bottom: 40px;
  padding: 0;
  z-index: 800;
  @include row--center;
}

.dataset-card__overlay__back {
  position: absolute;
  @include fullsize;
  background: linear-gradient(118.32deg, #2C4573 0%, #0B2448 83.62%);
  mix-blend-mode: darken;
  border-radius: 5px 5px 0 0;
  opacity: 0.7;
}

.dataset-card__overlay__select {
  position: absolute;
  top: 15px;
  left: 15px;
  width: 20px;
  height: 20px;
  border-radius: 5px;
  background: transparent;
}

.dataset-card__overlay__open {
  width: 40%;
  height: 40px;
  @include row--center;
  background: transparent;
  border: 2px solid $colorWhite;
  border-radius: 5px;
  @include typography(md-1, headlines, 500);
  color: $colorWhite;
  text-align: center;
  letter-spacing: 0.5px;
  opacity: 1;
  cursor: pointer;
  z-index: 999;
  transition: background-color 200ms linear, color 200ms linear;
  @include noSelect;

  &:hover, &:active {
    background: $colorWhite;
    color: $colorSecondaryDark1;
  }
}

.dataset-card__overlay__more {
  position: absolute;
  top: 5px;
  right: 10px;
  width: 32px;
  padding: 10px 0;
  @include row--center;
  @include noSelect;

  &:active {
    opacity: 0.8;
  }
}

.dataset-card__overlay__more__img {
  width: 32px;
  height: 9px;
  object-fit: contain;
}

.dataset-card__menu {
  position: absolute;
  top: 30px;
  right: -60px;
  width: 150px;
  height: 40px;
  border-radius: 10px;
  overflow: hidden;
  box-shadow: 0px 20px 60px rgba(11, 36, 72, 0.2), 0px 1px 2px rgba(58, 78, 108, 0.05);
  @include col;
  z-index: 1000;
}

.dataset-card__menu--right {
  right: 5px;
}

.dataset-card__menu__item {
  flex: 1;
  padding: 11px 16px;
  @include typography(md, default);
  line-height: 18px;
  background: $colorWhite;
  color: $colorSecondaryLight;

  cursor: pointer;

  &:not(.dataset-card__menu__item--disabled):hover {
    font-weight: bold;
    color: $colorSecondaryDark1;
    background: $colorGriteDark2;
  }

  &.dataset-card__menu__item--disabled {
    cursor: auto;
    color: $colorSecondaryLight1;
  }
}

.dataset-card__menu__item--gray {
  background: $colorGriteDark;
}

.dataset-card__status {
  position: absolute;
  left: 0; bottom: 0;
  @include row--center;
  background: $colorAliceShade;
  border-radius: 0 2px 0 0;
  width: 90px;
  height: 28px;
  z-index: 999;
}

.dataset-card__status-icon {
  width: 18px;
  height: 18px;
  object-fit: contain;
}
.dataset-card__status-text {
  margin-left: 10px;
  @include typography(md-1, headlines, bold);
  color: $colorAliceNight;
}

.dataset-card__downside {
  flex: 1;
  background: $colorAliceShade;
  padding: 12px 20px 12px 20px;
  position: relative;
  overflow: hidden;
}

.dataset-card__name {
  @include ellipsis(2, lg, 22px);
  @include typography(md-1, default, bold);
  word-break: break-all;
  min-height: 36px;
  text-decoration: none;
  color: $colorSecondaryDark;
  margin: 0;
}

.dataset-card__stats {
  @include row--distributed;
  @include typography(md, default);
  line-height: 20px;

  /* Graple Light 2 */
  color: $colorSecondaryLight;

  strong {
    font-weight: bold;
    color: $colorSecondaryDark;
  }
}

.dataset-card__progress__bar {
  width: 100%;
  height: 6px;
  position: relative;
  background: $colorSecondaryLight2;
  border-radius: 0 0 5px 5px;
  overflow: hidden;
}

.dataset-card__progress__bar__filled {
  position: absolute;
  bottom: 0;
  height: 6px;
  background: $colorPrimaryLight1;
  border-radius: 0 0 0 5px;
}
</style>
