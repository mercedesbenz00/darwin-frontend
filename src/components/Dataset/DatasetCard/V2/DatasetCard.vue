<template>
  <div
    class="dataset-card__container"
    v-tooltip="!selectable ? 'The dataset is already used in another workflow' : undefined"
    :class="{
      'dataset-card__container--selected': selected,
      'dataset-card__container--unselectable': !selectable
    }"
  >
    <div class="dataset-connected__container">
      <thumbnails
        :data="dataset.thumbnails"
        :max="3"
        variant="big"
      />
      <p
        v-tooltip="dataset.name"
        class="dataset-connected__label"
      >
        {{ dataset.name }}
      </p>
      <div class="dataset-connected__attribute-wrapper">
        <attribute-stack>
          <template #primary-attribute>
            {{ itemCount }}
          </template>
          <template #secondary-attribute>
            Items
          </template>
        </attribute-stack>
        <attribute-stack>
          <template #primary-attribute>
            {{ dataset.num_classes||0 }}
          </template>
          <template #secondary-attribute>
            Classes
          </template>
        </attribute-stack>
        <attribute-stack>
          <template #primary-attribute>
            {{ dataset.num_annotations||0 }}
          </template>
          <template #secondary-attribute>
            Annotations
          </template>
        </attribute-stack>
      </div>
      <div class="dataset-connected__progress-wrapper">
        <progress-bar
          variant="active"
          :value="progressValue"
        />
        <p class="dataset-connected__progress-label">
          {{ progressPercentage }}%
        </p>
      </div>
    </div>
  </div>
</template>

<script lang='ts'>
import {
  computed,
  defineComponent,
  PropType
} from 'vue'

import AttributeStack from '@/components/Common/AttributeStack/AttributeStack.vue'
import ProgressBar from '@/components/Common/ProgressBar/ProgressBar.vue'
import Thumbnails from '@/components/Common/Thumbnails/Thumbnails.vue'
import { DatasetPayload } from '@/store/types'

import { DatasetCardProps } from './types'

export default defineComponent({
  name: 'DatasetCardV2',
  components: {
    Thumbnails,
    AttributeStack,
    ProgressBar
  },
  props: {
    dataset: {
      required: true,
      type: Object as PropType<DatasetPayload>
    },
    selectable: {
      required: false,
      type: Boolean,
      default: false
    },
    selected: {
      required: false,
      type: Boolean,
      default: false
    }
  },
  setup (props: DatasetCardProps) {
    const progressValue = computed(() => {
      return props.dataset.progress || 0
    })

    const progressPercentage = computed(() => {
      return Math.round(progressValue.value * 100)
    })

    const itemCount = computed(() => {
      return (props.dataset.num_images || 0) + (props.dataset.num_videos || 0)
    })

    return {
      progressValue,
      progressPercentage,
      itemCount
    }
  }
})
</script>

<style lang='scss' scoped>
.dataset-card__container {
  width: 100%;
  transition: all 100ms ease;

  display: inline-block;
  padding-bottom: 3px;
  background-color: #FFFFFF;
  border-radius: 12px;
  border: 1px solid $colorStrokeRaise;
  box-shadow: 0px 0px 1px rgba(0, 0, 0, 0.03), 0px 1px 2px rgba(16, 24, 40, 0.05);;
  cursor: pointer;

  &:hover {
    box-shadow: 0px 0px 0px 1.5px $colorCheckBoxFocus;
  }

  &--selected {
    box-shadow: 0px 0px 0px 1.5px $colorCheckBoxFocus;
  }

  &--unselectable {
    &:hover {
      box-shadow: none;
    }
    cursor: default;
    opacity: 0.5;
  }
}
.dataset-connected__container {
  display: grid;
  grid-template-rows: repeat(4, min-content);
  grid-row-gap: 6px;
  padding: 6px;
}

.dataset-connected__label {
  margin-left: 2px;
  @include typography(md-1, inter, 500);
  color: $colorContentDefault;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.dataset-connected__attribute-wrapper {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 2px;
}

.dataset-connected__progress-wrapper {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.dataset-connected__progress-label {
  @include typography(sm, inter, 500);
  color: $colorContentTertiary;
  text-align: right;
  margin-left: 10px;
}
</style>
