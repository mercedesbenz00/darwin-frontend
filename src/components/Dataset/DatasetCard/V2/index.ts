export * from './types'

export { default as DatasetCardV2 } from './DatasetCard.vue'
