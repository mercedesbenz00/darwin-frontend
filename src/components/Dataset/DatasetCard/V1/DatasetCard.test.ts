import { createLocalVue, shallowMount, Stubs } from '@vue/test-utils'
import flushPromises from 'flush-promises'
import VueLazyload from 'vue-lazyload'

import { buildDatasetPayload } from 'test/unit/factories'
import { Transition } from 'test/unit/stubs'

import DatasetCard from '@/components/Dataset/DatasetCard/V1/DatasetCard.vue'
import clickOutsideDirective from '@/directives/click-outside'
import { DatasetPayload } from '@/store/types'

const localVue = createLocalVue()
localVue.use(VueLazyload)
localVue.directive('tooltip', () => {})
localVue.directive('click-outside', clickOutsideDirective)

const dataset = buildDatasetPayload({
  id: 1,
  owner_id: 1,
  active: true,
  thumbnails: ['Image 1', 'Image 2', 'Image 3', 'Image 4', 'Image 5', 'Image 6'],
  work_size: 30,
  slug: 'smart-fume-hood-segmentation-and-direction',
  public: false,
  progress: 0.0,
  num_images: 16,
  num_videos: 1,
  num_classes: 6,
  num_annotations: 1000,
  name: 'Smart Fume Hood Segmentation and Direction',
  inserted_at: '2019-08-12T17:56:54',
  updated_at: '2019-08-12T17:56:54'
})

let propsData: {
  data: DatasetPayload,
  menuAlign?: string,
  disableMenu?: boolean,
  selectable?: boolean,
  selected?: boolean
}

const stubs: Stubs = { 'router-link': true, Transition }
const mocks = { $can: () => true }

beforeEach(() => {
  propsData = { data: dataset }
})

it('matches snapshot', async () => {
  const wrapper = shallowMount(DatasetCard, { localVue, propsData, stubs, mocks })
  await flushPromises()
  expect(wrapper).toMatchSnapshot()
})

it('matches snapshot when selected', async () => {
  propsData.selectable = true
  propsData.selected = true
  const wrapper = shallowMount(DatasetCard, { localVue, propsData, stubs, mocks })
  await flushPromises()
  expect(wrapper).toMatchSnapshot()
})

it('matches snapshot when menu disabled', async () => {
  propsData.disableMenu = true
  const wrapper = shallowMount(DatasetCard, { localVue, propsData, stubs, mocks })
  await flushPromises()
  expect(wrapper).toMatchSnapshot()
})

it('renders the link correctly for version 1', async () => {
  const wrapper = shallowMount(DatasetCard, { localVue, propsData, stubs, mocks })
  await flushPromises()
  expect(wrapper.find('.dataset-card').attributes('to')).toContain(`/datasets/${dataset.id}/dataset-management`)
})

it('renders the link for excluding archive items for other versions', async () => {
  propsData.data.version = 2
  const wrapper = shallowMount(DatasetCard, { localVue, propsData, stubs, mocks })
  await flushPromises()
  expect(wrapper.find('.dataset-card').attributes('to')).toContain(`/datasets/${dataset.id}/dataset-management?not_statuses=archived`)
})

it('allows deletion through overlay menu', async () => {
  const wrapper = shallowMount(DatasetCard, { localVue, propsData, stubs, mocks })
  await flushPromises()

  await wrapper.trigger('mouseover')
  await wrapper.find('.dataset-card__overlay__more').trigger('click')
  expect(wrapper.find('.dataset-card__menu__item').exists()).toBe(true)

  await wrapper.find('.dataset-card__menu__item:last-child').trigger('click')
  expect(wrapper.emitted().delete).toEqual([[dataset]])
})

it('matches snapshot when progress > 0%', async () => {
  propsData.data.progress = 0.5
  const wrapper = shallowMount(DatasetCard, { localVue, propsData, stubs, mocks })
  await flushPromises()
  expect(wrapper).toMatchSnapshot()
})
