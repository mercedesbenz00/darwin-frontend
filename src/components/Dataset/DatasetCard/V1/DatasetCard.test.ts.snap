// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<router-link-stub to="/datasets/1/dataset-management" class="dataset-card">
  <div class="dataset-card__upside">
    <div class="dataset-card__upside__row dataset-card__upside__row--top">
      <div class="dataset-card__img"><img data-src="Image 1" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
      <div class="dataset-card__img"><img data-src="Image 2" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
      <div class="dataset-card__img"><img data-src="Image 3" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
    </div>
    <div class="dataset-card__upside__row">
      <div class="dataset-card__img"><img data-src="Image 4" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
      <div class="dataset-card__img"><img data-src="Image 5" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
      <div class="dataset-card__img"><img data-src="Image 6" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
    </div>
    <transition-stub name="fade">
      <div class="dataset-card__overlay">
        <div class="dataset-card__overlay__more"><img src="/static/imgs/more-white-icon.svg" class="dataset-card__overlay__more__img"></div>
      </div>
    </transition-stub>
    <!---->
    <!---->
  </div>
  <div class="dataset-card__downside">
    <div class="dataset-card__name">
      Smart Fume Hood Segmentation and Direction
    </div>
    <div class="dataset-card__stats">
      <div><strong>17</strong> Files
      </div>
      <div><strong>6</strong> Classes
      </div>
      <div><strong>0%</strong> Complete
      </div>
    </div>
  </div>
  <div class="dataset-card__progress__bar">
    <div class="dataset-card__progress__bar__filled" style="width: 0%;"></div>
  </div>
</router-link-stub>
`;

exports[`matches snapshot when menu disabled 1`] = `
<router-link-stub to="/datasets/1/dataset-management" class="dataset-card">
  <div class="dataset-card__upside">
    <div class="dataset-card__upside__row dataset-card__upside__row--top">
      <div class="dataset-card__img"><img data-src="Image 1" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
      <div class="dataset-card__img"><img data-src="Image 2" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
      <div class="dataset-card__img"><img data-src="Image 3" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
    </div>
    <div class="dataset-card__upside__row">
      <div class="dataset-card__img"><img data-src="Image 4" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
      <div class="dataset-card__img"><img data-src="Image 5" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
      <div class="dataset-card__img"><img data-src="Image 6" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
    </div>
    <!---->
    <!---->
    <!---->
  </div>
  <div class="dataset-card__downside">
    <div class="dataset-card__name">
      Smart Fume Hood Segmentation and Direction
    </div>
    <div class="dataset-card__stats">
      <div><strong>17</strong> Files
      </div>
      <div><strong>6</strong> Classes
      </div>
      <div><strong>0%</strong> Complete
      </div>
    </div>
  </div>
  <div class="dataset-card__progress__bar">
    <div class="dataset-card__progress__bar__filled" style="width: 0%;"></div>
  </div>
</router-link-stub>
`;

exports[`matches snapshot when progress > 0% 1`] = `
<router-link-stub to="/datasets/1/dataset-management?not_statuses=archived" class="dataset-card">
  <div class="dataset-card__upside">
    <div class="dataset-card__upside__row dataset-card__upside__row--top">
      <div class="dataset-card__img"><img data-src="Image 1" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
      <div class="dataset-card__img"><img data-src="Image 2" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
      <div class="dataset-card__img"><img data-src="Image 3" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
    </div>
    <div class="dataset-card__upside__row">
      <div class="dataset-card__img"><img data-src="Image 4" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
      <div class="dataset-card__img"><img data-src="Image 5" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
      <div class="dataset-card__img"><img data-src="Image 6" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
    </div>
    <transition-stub name="fade">
      <div class="dataset-card__overlay">
        <div class="dataset-card__overlay__more"><img src="/static/imgs/more-white-icon.svg" class="dataset-card__overlay__more__img"></div>
      </div>
    </transition-stub>
    <!---->
    <!---->
  </div>
  <div class="dataset-card__downside">
    <div class="dataset-card__name">
      Smart Fume Hood Segmentation and Direction
    </div>
    <div class="dataset-card__stats">
      <div><strong>17</strong> Files
      </div>
      <div><strong>6</strong> Classes
      </div>
      <div><strong>50%</strong> Complete
      </div>
    </div>
  </div>
  <div class="dataset-card__progress__bar">
    <div class="dataset-card__progress__bar__filled" style="width: 50%;"></div>
  </div>
</router-link-stub>
`;

exports[`matches snapshot when selected 1`] = `
<div to="/datasets/1/dataset-management" class="dataset-card dataset-card--selected">
  <div class="dataset-card__upside">
    <div class="dataset-card__upside__row dataset-card__upside__row--top">
      <div class="dataset-card__img"><img data-src="Image 1" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
      <div class="dataset-card__img"><img data-src="Image 2" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
      <div class="dataset-card__img"><img data-src="Image 3" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
    </div>
    <div class="dataset-card__upside__row">
      <div class="dataset-card__img"><img data-src="Image 4" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
      <div class="dataset-card__img"><img data-src="Image 5" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
      <div class="dataset-card__img"><img data-src="Image 6" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" lazy="loading"></div>
    </div>
    <transition-stub name="fade">
      <div class="dataset-card__overlay">
        <div class="dataset-card__overlay__more"><img src="/static/imgs/more-white-icon.svg" class="dataset-card__overlay__more__img"></div>
      </div>
    </transition-stub>
    <!---->
    <!---->
  </div>
  <div class="dataset-card__downside">
    <div class="dataset-card__name">
      Smart Fume Hood Segmentation and Direction
    </div>
    <div class="dataset-card__stats">
      <div><strong>17</strong> Files
      </div>
      <div><strong>6</strong> Classes
      </div>
      <div><strong>0%</strong> Complete
      </div>
    </div>
  </div>
  <div class="dataset-card__progress__bar">
    <div class="dataset-card__progress__bar__filled" style="width: 0%;"></div>
  </div>
</div>
`;
